test/viewmodels/home_view_model_unit_test.dart

```diff
@@ -0,0 +1,295 @@
+            publicationTime: element.publicationTime,
+            voteScore: element.voteScore,
+          ),
+        );
+
+        final posts = postsData.map((data) {
+          return MockPostFirestore.createPostAt(data, point);
+        }).toList();
+
+        final expectedPosts = postsData.mapIndexed(
+          (index, data) {
+            return PostOverview(
+              title: data.title,
+              description: data.description,
+              votes: data.voteScore,
+              commentNumber: 0,
+              posterUsername: owners[index].data.username,
+            );
+          },
+        );
+
+        // Mock the repository calls
+        for (var i = 0; i < numberOfPosts; i++) {
+          when(userRepository.getUser(posts[i].data.ownerId)).thenAnswer(
+            (_) async => owners[i],
+          );
+        }
+
+        when(geoLocationService.getCurrentPosition()).thenAnswer(
+          (_) async => point,
+        );
+        when(postRepository.getNearPosts(point, HomeViewModel.kmPostRadius))
+            .thenAnswer(
+          (_) async => posts,
+        );
+
+        // Check the actual posts
+        final actualPosts = await container.read(postOverviewProvider.future);
+
+        expect(actualPosts, expectedPosts);
+      },
+    );
+
+    test("New posts are exposed correctly on refresh", () async {
+      when(geoLocationService.getCurrentPosition()).thenAnswer(
+        (_) async => point,
+      );
+      when(postRepository.getNearPosts(point, HomeViewModel.kmPostRadius))
+          .thenAnswer(
+        (_) async => [],
+      );
+
+      // Query the posts a first time
+      final postBeforeRefresh =
+          await container.read(postOverviewProvider.future);
+      expect(postBeforeRefresh, []);
+
+      // Simulate a new post being added
+      final owner = MockUserFirestore.generateUserFirestore(1)[0];
+      final postData = MockPostFirestore.generatePostData(1)
+          .map(
+            (postData) => PostData(
+              ownerId: owner.uid,
+              title: postData.title,
+              description: postData.description,
+              publicationTime: postData.publicationTime,
+              voteScore: postData.voteScore,
+            ),
+          )
+          .toList()[0];
+      final post = MockPostFirestore.createPostAt(postData, point);
+
+      when(userRepository.getUser(post.data.ownerId)).thenAnswer(
+        (_) async => owner,
+      );
+
+      final expectedPosts = [
+        PostOverview(
+          title: post.data.title,
+          description: post.data.description,
+          votes: post.data.voteScore,
+          commentNumber: 0,
+          posterUsername: owner.data.username,
+        ),
+      ];
+
+      when(postRepository.getNearPosts(point, HomeViewModel.kmPostRadius))
+          .thenAnswer(
+        (_) async => [post],
+      );
+      when(userRepository.getUser(post.data.ownerId)).thenAnswer(
+        (_) async => owner,
```

Comment:
Code duplication with lines 247 and 248
```suggestion
```

Reply:
Correct !
I remove the duplication in 8330ba2973f4186eedd20abf0958c0a4a793e912.